app/src/main/java/org/p2p/wallet/user/interactor/UserTokensInteractor.kt

```diff
@@ -0,0 +1,18 @@
+package org.p2p.wallet.user.interactor
+
+import timber.log.Timber
+import org.p2p.core.token.Token
+import org.p2p.token.service.api.events.manager.TokenServiceEventPublisher
+import org.p2p.token.service.model.TokenServiceNetwork
+
+class UserTokensInteractor(private val tokenServiceInteractor: TokenServiceEventPublisher) {
+
+    fun loadUserRates(userTokens: List<Token.Active>) {
+        Timber.i("Loading user rates for ${userTokens.size}")
+        val tokenAddresses = userTokens.map { it.mintAddress }
+        tokenServiceInteractor.loadTokensPrice(
```

Comment:
and why EventPublisher is responsible for loading token prices? we have repository for that